source/common/quic/envoy_quic_utils.h

```diff
@@ -105,14 +105,22 @@ quicHeadersToEnvoyHeaders(const quic::QuicHeaderList& header_list, HeaderValidat
 }
 
 template <class T>
-std::unique_ptr<T> spdyHeaderBlockToEnvoyHeaders(const spdy::SpdyHeaderBlock& header_block) {
+std::unique_ptr<T> spdyHeaderBlockToEnvoyHeaders(const spdy::SpdyHeaderBlock& header_block,
```

Comment:
I just noticed that we are not validating trailers values. Does it make more sense to pass in HeaderValidator and error details to this helper function as quicHeadersToEnvoyHeaders() does?

Reply:
Good call - done!